File: src/main/java/xyz/capybara/clamav/commands/scan/MultiScan.java
package xyz.capybara.clamav.commands.scan;

import xyz.capybara.clamav.commands.Command;

import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;

public class MultiScan extends ScanCommand {

    private String path;

    public MultiScan(String path) {
        this.path = path;
    }

    @Override
    public String getCommandString() {
        return "MULTISCAN";
    }

    @Override
    protected Command.CommandFormat getFormat() {
        return Command.CommandFormat.NEW_LINE;
    }

    @Override
    protected ByteBuffer getRawCommand() {
        StringBuilder rawCommand = new StringBuilder();
        rawCommand.append(getFormat().getPrefix())
                .append(getCommandString())
                .append(' ')
                .append(path)
                .append(getFormat().getTerminator());

        return ByteBuffer.wrap(rawCommand.toString().getBytes(StandardCharsets.US_ASCII));
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/scan/MultiScan.kt
package xyz.capybara.clamav.commands.scan

import xyz.capybara.clamav.commands.Command

import java.nio.ByteBuffer
import java.nio.charset.StandardCharsets

internal class MultiScan(private val path: String) : ScanCommand() {
    override val commandString
        get() = "MULTISCAN"

    override val format
        get() = CommandFormat.NEW_LINE

    override val rawCommand: ByteBuffer
        get() = ByteBuffer.wrap("${format.prefix}$commandString $path${format.terminator}".toByteArray(StandardCharsets.US_ASCII))
}
